src/plugins/kibana_usage_collection/server/collectors/event_loop_delays/rollups/daily.test.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { rollDailyData } from './daily';
+import { loggingSystemMock, savedObjectsRepositoryMock } from '../../../../../../core/server/mocks';
+import type { SavedObjectsFindResponse } from '../../../../../../core/server';
+
+describe('rollDailyData', () => {
+  const logger = loggingSystemMock.createLogger();
+  const mockSavedObjectsClient = savedObjectsRepositoryMock.create();
+
+  beforeEach(() => jest.clearAllMocks());
+
+  it('returns false if no savedObjectsClient', async () => {
+    const result = await rollDailyData(logger, undefined);
+    expect(result).toBe(false);
+    expect(mockSavedObjectsClient.find).toBeCalledTimes(0);
+  });
+
+  it('calls delete on documents older than 3 days', async () => {
```

Comment:
Maybe it should be tested in an integration test? Tests with assertions against actions (`delete`) do not verify the final state is in the desired shape.